src/Components/Components/src/Lifetime/IComponentApplicationStateStore.cs

```diff
@@ -0,0 +1,24 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components
```

Comment:
Do these types need to appear in a top level namespace?

Reply:
We can move this, the only one I want in the top level is ComponentApplicationState since that's what the components and services interact with.

I'll move this and others to the Lifetime namespace.